### ISSUE TITLE
Port New Ruby Problem Solving Lesson to Javascript Course

### ISSUE THREAD

**Issue by KevinMulhern** [TheOdinProject/curriculum#16122]
**Description:**
We have recently added a problem solving lesson to the ruby course which would be great for the JS course as well with a few modifications. 

Link to the lesson file: https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/basic_ruby/problem_solving.md

**Acceptance Criteria:**
1. Change the Fizz Buzz Code Examples to use JS.
2. Add the Lesson to somewhere in the js course (check the comments for where)

**Comment by KevinMulhern** [comment]
@codyloyd where do you reckon we should put this in the JS course?

**Comment by ZachBaird** [comment]
I can start working on this @KevinMulhern @I3uckwheat . Idk where in the full-stack JS track it should go, but it might be beneficial (yet repetitive) to put it in the JS section all tracks see.

**Comment by marvingay** [comment]
As currently laid out, it would be best put before [a quick review](https://github.com/TheOdinProject/curriculum/blob/master/javascript/introduction/js101-review.md), but as noted, that would cause a repetition issue. Unless there is a way to conditionally show it based on the user's track.

**Comment by I3uckwheat** [comment]
@ZachBaird 

I've assigned it to you :)

I'm not sure where it would go juuust yet. But it looks like this could be simplified to pseudo-code and made generic instead. 

Or, we could use the same [technique as this](https://www.theodinproject.com/courses/web-development-101/lessons/prerequisites) and put it in both places.

**Comment by ZachBaird** [comment]
@I3uckwheat Cool :) Lemme know when a decision is made and I'll get it in there.

Taking in @marvingay 's recommendation, I quickly clicked through NodeJS and didn't really see anywhere it would make sense to put this. However, maybe we could put this where Marvin mentions and then move it to NodeJS whenever the Computer Science lesson gets replicated from Ruby to Node.

**Comment by I3uckwheat** [comment]
@codyloyd Do you have any suggestions on this one?

**Comment by ZachBaird** [comment]
@I3uckwheat @KevinMulhern @codyloyd 
Have you guys given any more thought where this could go? I've got a first draft ready for review and I plopped it into javascript > introduction > problem_solving for now after the introduction. My thinking was, for the time being, it could be a "now before we hop into the good stuff..." kind of deal.

If any ideas aren't forthcoming, I can at least put up a WIP pull request for review until we know where to put it.
